### ISSUE TITLE
Safe directory issue with git 2.25.1 on Ubuntu 20.04

### ISSUE THREAD

**Issue by JulianWgs** [airspeed-velocity/asv#1212]
When using asv this happens:

```bash
$ asv run       
· Creating environments.
· Discovering benchmarks
·· Error running /usr/bin/git checkout -f 2cc467b7d4830daafc6cab8af8b61a2979ea9dce (exit status 128)
   STDOUT -------->
   
   STDERR -------->
   fatal: unsafe repository ('/XXX/.asv/env/76321772e92136ec84b9940d7022132a/project' is owned by someone else)
   To add an exception for this directory, call:
   
   	git config --global --add safe.directory /XXX/.asv/env/76321772e92136ec84b9940d7022132a/project

·· Failed to build the project and import the benchmark suite.
```

I'm using git 2.25.1 (which is the newest on Ubuntu 20.04) and asv 0.5.1.

I assume the error happens in reaction to [this git vulnerability](https://github.blog/2022-04-12-git-security-vulnerability-announced/). It may be a fix specific to Ubuntu, since I could't find a mention in the git release notes. I currently don't have the time to find a workaround/fix. Just wanted to report this for now.

If this is not an issue of the asv project I will also happily create a Stack Overflow question.

**Comment by datapythonista** [comment]
Did you check if the user who owns the .git directory is the same as the current directory and the current user? I think what's happening is correct to avoid the vulnerability you link to. I think git does the right thing, and asv doesn't want to make git unsafe.

**Comment by JulianWgs** [comment]
Yes, you are correct! The data is on a disk which is NTFS formatted, therefore the owners don't match up even though my user created the files.

Moving the folder to an Ext4 partition solves the problem. Thanks!

Sorry, for the inconvenience :)
